lightstep/recorder.py

```diff
@@ -99,13 +99,26 @@ def __init__(self,
         atexit.register(self.shutdown)
 
         self._periodic_flush_seconds = periodic_flush_seconds
-        self._flush_connection = conn._Connection(self._collector_url)
-        self._flush_connection.open()
+        # _flush_connection and _flush_thread are created lazily since some
+        # Python environments (e.g., Tornado) fork() initially and mess up the
+        # reporting machinery up otherwise.
+        self._flush_connection = None
+        self._flush_thread = None
         if self._periodic_flush_seconds <= 0:
             warnings.warn(
                 'Runtime(periodic_flush_seconds={0}) means we will never flush to lightstep unless explicitly requested.'.format(
                     self._periodic_flush_seconds))
-        else:
+
+    def _maybe_init_flush_thread(self):
+        """Start a periodic flush mechanism for this recorder if:
+
+        1. periodic_flush_seconds > 0, and 
+        2. self._flush_thread is None *or* the pid has changed (indicating a
+           fork() of some sort)
```

Comment:
@ross this was an oversight. I was previously doing the `os.getpid()` thing here, but the more I thought about it the more I realized how insufficient the local check was against arbitrary `fork()` calls. Hence the scaled-back version that I have here.

@jmacd yes, exactly. And this is an idiom used by `tornado` at startup (as well as other python server frameworks like `gunicorn`).

Anyway, I've updated the comment to match the code. After that I'll merge and push out a new artifact.